packages/amplify-provider-awscloudformation/lib/amplify-service-manager.js

```diff
@@ -0,0 +1,344 @@
+    } catch (e) {
+      // do nothing
+    }
+
+    if (teamProviderFilePath && fs.existsSync(teamProviderFilePath)) {
+      const teamProviderInfo = context.amplify.readJsonFile(teamProviderFilePath);
+
+      const envList = Object.keys(teamProviderInfo);
+
+      let appIdsInTheSameLocalProjectAndRegion = [];
+      for (let i = 0; i < envList.length; i++) {
+        const env = envList[i];
+        if (
+          env !== envName &&
+          teamProviderInfo[env][constants.ProviderName].Region === awsConfig.region &&
+          teamProviderInfo[env][constants.ProviderName][constants.AmplifyAppIdLabel]
+        ) {
+          appIdsInTheSameLocalProjectAndRegion.push(teamProviderInfo[env][constants.ProviderName][constants.AmplifyAppIdLabel]);
+        }
+      }
+
+      if (appIdsInTheSameLocalProjectAndRegion.length > 0) {
+        let apps = [];
+        let listAppsResponse = {};
+        do {
+          listAppsResponse = await amplifyClient
+            .listApps({
+              nextToken: listAppsResponse.nextToken,
+              maxResults: 25,
+            })
+            .promise();
+          apps = apps.concat(listAppsResponse.apps);
+        } while (listAppsResponse.nextToken);
+
+        const verifiedAppIds = apps.map(app => app.appId);
+        appIdsInTheSameLocalProjectAndRegion = appIdsInTheSameLocalProjectAndRegion.filter(appId => verifiedAppIds.includes(appId));
+
+        if (appIdsInTheSameLocalProjectAndRegion.length === 1) {
+          amplifyAppId = appIdsInTheSameLocalProjectAndRegion[0]; // eslint-disable-line
+        } else if (appIdsInTheSameLocalProjectAndRegion.length > 1) {
+          context.print.info(`You project is associated with multiple Amplify Service Apps in the region ${awsConfig.region}`);
+          amplifyAppId = await SelectFromExistingAppId(context, appIdsInTheSameLocalProjectAndRegion);
+        }
+      }
+    }
+  }
+
+  if (!amplifyAppId) {
+    const createAppParams = {
+      name: projectName,
+    };
+    const createAppResponse = await amplifyClient.createApp(createAppParams).promise();
+    amplifyAppId = createAppResponse.app.appId;
+  }
+
+  let verifiedStackName = stackName;
+  let deploymentBucketName = `${stackName}-deployment`;
+
+  let needToCreateNewBackendEnv = false;
+  try {
+    const { backendEnvironment } = await amplifyClient
+      .getBackendEnvironment({
+        appId: amplifyAppId,
+        environmentName: envName,
+      })
+      .promise();
+
+    if (backendEnvironment) {
+      verifiedStackName = backendEnvironment.stackName;
+      deploymentBucketName = backendEnvironment.deploymentArtifacts;
+    } else {
+      needToCreateNewBackendEnv = true;
+    }
+  } catch (e) {
+    needToCreateNewBackendEnv = true;
+  }
+
+  if (needToCreateNewBackendEnv) {
+    context.print.info(`New Amplify Service backend environemnt ${envName} will be created under project: ${amplifyAppId}`);
```

Comment:
Adding backend environment ${envName} to AWS Amplify Console app: ${**amplifyAppName**}

Reply:
Changed. 
but we don't have variable `amplifyAppName` averrable here, using amplifyAppId